design/pv_backup_info.md

```diff
@@ -0,0 +1,174 @@
+    RetainedSnapshot string    // The name or ID of the snapshot associated object(SAO).
+}
+
+// VeleroNativeSnapshotInfo is used for displaying the Velero native snapshot status.
+type VeleroNativeSnapshotInfo struct {
+    VolumeType string    // The cloud provider snapshot volume type.
+    VolumeAZ   string    // The cloud provider snapshot volume's availability zones.
+    IOPS       string    // The cloud provider snapshot volume's IOPS.
+}
+
+// PodVolumeBackupInfo is used for displaying the PodVolumeBackup snapshot status.
+type PodVolumeBackupInfo struct {
+    UploaderType  string    // The type of the uploader that uploads the data. The valid values are `kopia` and `restic`. It's useful for file-system backup and snapshot data mover.
+    VolumeName    string // The PVC's corresponding volume name used by Pod
+    PodName       string // The Pod name mounting this PVC. The format should be <namespace-name>/<pod-name>.
+}
+```
+
+To make Velero support multiple versions of VolumeInfo, Velero needs to create a structure to include the version and the versioned volume information together. The information is persisted in the metadata file during backup creation. When reading the VolumeInfo metadata file, Velero reads the version information first by un-marshalling the metadata file by structure `VolumeInfoVersion`, then decide to use which version of the VolumeInfos according to the version value.
+
+If there is need to introduce a non compatible change to the VolumeInfo, a new version of `VolumeInfos` and `VolumeInfo` are needed. For example, version 2.0 is created, then `VolumeInfosV20` and `VolumeInfoV20` structures are needed.
+
+``` golang
+type VolumeInfoVersion struct {
+    Version string 
+}
+
+type VolumeInfosV10 struct {
+    Infos   []VolumeInfoV10
+    Version string // VolumeInfo structure's version information.
+}
+```
+
+### How the VolumeInfo array is generated.
+
+A new methods are added to BackupStore to download the VolumeInfo metadata file.
+Uploading the metadata file is covered in the exiting `PutBackup` method.
+
+``` golang
+type BackupStore interface {
+    ...
+    GetVolumeInfos(name string) ([]*VolumeInfo, error)
+    ...
+}
+```
+
+The VolumeInfo array is generated before the backup is persisted by this function `persistBackup`. Then the VolumeInfo array will be passed to `persistBackup` function as a parameter.
+
+Please notice that the change happened in async operations are not reflected in the new metadata file. The file only covers the volume changes happen in the Velero server process scope.
+
+### How the VolumeInfo array is used.
+
+#### Generate the PVC backed-up information summary
+The upstream tools can use this VolumeInfo array to format and display their volume information. This is in the scope of this feature.
+
+#### Retrieve volume backed-up information for `velero backup describe` command
```

Comment:
Just to clarify in v1.13 we only read the pvInfo array proposed in this doc when necessary, and it should be only to address new use cases right?

Do we plan to re-write any code in `backup/restore describe` or the func `restoreItem`?

Reply:
`restoreItem` is in the scope of v1.13 because it's used to address issue #6595.
`backup/restore describe` is not in the plan yet.